pydantic/_internal/_dataclasses.py

```diff
@@ -76,9 +78,9 @@ def prepare_dataclass(
         warning_string = (
             f'`{name}` is not fully defined, you should define `{e}`, then call TODO! `methods.rebuild({name})`'
         )
-        if config['undefined_types_warning']:
+        if config['undefined_types_warning']:  # type: ignore
```

Comment:
I don't love that we are now type-ignoring every access to `Config`. feels like we should either use `.get` or tweak the config implementation. We don't need to do that as part of this PR but we probably should do it sooner rather than later if we merge this.

Reply:
Totally agreed. This is just making explicit that what we're currently doing is questionable.

Reply:
we absolutely need to change how config works, I'll work on that today.